src/components/layout/PanelChange/index.tsx

```diff
@@ -74,12 +79,20 @@ export const PanelChange: FC<IProps> = ({
   const [value, setValue] = useState('');
   const [isLoading, setIsLoading] = useState(true);
   const [lastDistribution, setLastDistribution] = useState<Date>();
+  const [shareScaler, setShareScaler] = useState(1e3);
   // const { t } = useTranslation('main');
 
   useEffect(() => {
     setIsLoading(mainLoading);
   }, [mainLoading]);
 
+  useEffect(() => {
+    if (web3?.currentProvider === null || flowType !== FlowTypes.market) return;
+    getShareScaler(web3, exchangeKey, tokenA, tokenB).then((res) => {
```

Comment:
Take careful here, maybe there's memory leak.

Reply:
[how to fix memory leak issue in react hooks](https://medium.com/technology-hits/how-to-fix-memory-leak-issue-in-react-js-using-hook-a5ecbf9becf8)

Reply:
Thanks, added a boolean to prevent those sneaky memory leaks 👍